### PR TITLE
Tests on synthetic images

### PR DESCRIPTION
**Pull request by tback**
This implements tests on synthetic images to detect corner cases. WebP shrink on load breaks some of these - this is the reason my pull requests is based on 1.0.14.

### CODE REVIEW

**Comment by tback** [comment]
The tests are failing because travis runs the tests against a commit that is merged into the master. As I said, 

> WebP shrink on load breaks some of these - this is the reason my pull requests is based on 1.0.14

and this is exactly what travis shows now.

@greut do you think you'll get those green?

File: resizer_test.go [hunk 1]
```diff
@@ -136,7 +136,7 @@ func TestResizeSyntheticImages(t *testing.T) {
 			DifferingExpectations{}},
 		{"synthetic_horizontal.jpg",
 			Options{Crop: true, Width: 500, Height: 1000},
-			DifferingExpectations{}}, // unclear why this is 500 x 1000 when
+			DifferingExpectations{}}, // unclear why this is not 500 x 20 but vips-thumbnail does the same
```
Comment (jayme-github) [inline resizer_test.go]: One would expect thos to be 500x20 because `Enlarge` is `false`.

**Comment by jayme-github** [comment]
For reference, some text about resize operations and what to expect from them: https://github.com/h2non/bimg/issues/46

**Comment by tback** [comment]
I'll need a decision on what exactly is bimg's expected behavior in these cases.
